app.py

```diff
@@ -1,25 +1,104 @@
+from flask_sqlalchemy import SQLAlchemy
+from flask import session
+
+load_dotenv(find_dotenv())
 
 app = flask.Flask(__name__)
 
-# set up a separate route to serve the index.html file generated
-# by create-react-app/npm run build.
-# By doing this, we make it so you can paste in all your old app routes
-# from Milestone 2 without interfering with the functionality here.
 bp = flask.Blueprint(
     "bp",
     __name__,
     template_folder="./static/react",
 )
 
-# route for serving React page
-@bp.route("/")
-def index():
-    # NB: DO NOT add an "index.html" file in your normal templates folder
-    # Flask will stop serving this React page correctly
-    return flask.render_template("index.html")
+app.config["SECRET_KEY"] = "I don't have a secret key, wizard!"
+# Point SQLAlchemy to your Heroku database
+app.config["SQLALCHEMY_DATABASE_URI"] = os.getenv("DATABASE_URL")
+# Gets rid of a warning
+app.config["SQLALCHEMY_TRACK_MODIFICATIONS"] = False
+
+db = SQLAlchemy(app)
+
+login_manager = LoginManager(app)
+login_manager.login_view = "app.login"
+login_manager.init_app(app)
+
+
+@login_manager.user_loader
+def load_user(user_id):
+    """This function is used to load user"""
+    # since the user_id is just the primary key of our user table, use it in the query for the user
+    return User.query.get(int(user_id))
+
+
+class User(db.Model, UserMixin):
+    """This class is used to create User database"""
+
+    __tablename__ = "users"
+    id = db.Column(db.Integer(), primary_key=True)
+    username = db.Column(db.String(50), nullable=False, unique=True)
+    password = db.Column(db.String(100), nullable=False)
+
+db.create_all()
+
+
+@app.route("/")
+def start():
+    """This function is called on landing page to sign up"""
+    value = 0
+    page = 0
+    if "page" in session:
+        page = session["page"]
+    if "user_exists" in session:
+        if session["user_exists"]:
+            value = 1
+    return flask.render_template("signup.html", value=value, page=page)
+
+
+@app.route("/signUpPage", methods=["GET", "POST"])
+def sign():
+    """This function is called on signUpPage to store the form data"""
+    # code to validate and add user to database goes here
+    username = flask.request.form.get("username")
+    password = flask.request.form.get("password")
+    # if this returns a user, then the username already exists in database
+    user = User.query.filter_by(username=username).first()
+
+    if user:
+        # if a user is found, we want to redirect back to signup page so user can try again
+        session["user_exists"] = True
+        session["page"] = 1
+        return flask.redirect(flask.url_for("start"))
+
+    # create a new user with the form data. Hash the password so the plaintext version isn't saved.
+    new_user = User(username=username, password=password)
+
+    # add the new user to the database
+    db.session.add(new_user)
+    db.session.commit()
```

Comment:
Shouldn't this only be called if a form posts to this page? What if the person want's to login without signing up, as they have already signed up beforehand?